### PR TITLE
Presign url support

### PR DESCRIPTION
**Pull request by sandeep-paliwal**
Added support to generate signature to access private storage files
TVM will take following params to generate signature in a norma request
1) presignReq=true
2) blobName=<name of file to generate signature for>
3) expiryInSeconds=<how long the signature remains valid>
## How Has This Been Tested?

unit tests and manual TVM deployment testing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
- [x ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

### CODE REVIEW

File: lib/Tvm.js [hunk 1]
```diff
@@ -202,7 +229,7 @@ class Tvm {
       // appear in the I/O Runtime activation logs as an application error
       return {
         error: {
-          body: { error: 'server error' },
+          body: { error: 'server error ' + e.message },
```
Comment (moritzraho) [inline lib/Tvm.js:232]: let's not expose any uncatched error message for security reasons to the user, either we catch it and know what to return to the user either we don't know and we should look at the logs.

File: lib/Tvm.js [hunk 2]
```diff
@@ -36,6 +36,14 @@ class Tvm {
       // owAuth is redundant here as we already check it but it cannot hurt to check twice
       owAuth: joi.string().required()
     }
+
+    const tempSchema = {
```
Comment (moritzraho) [inline lib/Tvm.js:40]: that schema is specific to azure blob signature, it should be set elsewhere, the class is global to all endpoints

**Comment by meryllblanchet** [comment]
Adding @shazron as reviewer during @moritzraho 's time off.

File: lib/Tvm.js (hunk 1 above)
Comment (shazron) [reply lib/Tvm.js:232]: just show a generic error message.

**Comment by sandeep-paliwal** [comment]
@shazron  I have updated the PR based on some changes I had discussed with @moritzraho 
changes you suggested will also get addressed with this.

**Pull request merged**